feat: Post Aggregate Mock Api

yscarlet opened the pull request:
## 🐣Title
feat: post 기능 구현

## 🐣Part
BE - post

## 🐣Key Changes
domain 패키지 - Entity 설계
dao 패키지 - repository 작성
dto 패키지
api 패키지 - controller 개발
application - 하는듕..

## 🐣Next
service 로직 구현
api연결
comment 연결법 생각
like, jinji, scrap 처리 과정 생각

## 🐣질문
- DTO를 작성해야하는 그 기준이 궁금합니다.
가령 responsedto들은 그 형태가 같은 것들이 많은데 다 따로 만들어야 하나요?
- 말고도 질문이 23864263754개쯤 되는데 아직 뇌에서 정리가 안되서 정리 되면 여쭤봐도 되나요
- 훈수? 말고 혼냄?도 괜찮습니다. 코드 이상하게 짜는 게 더 싫어요 이미 이상한 것 같지만

## 🐣Issue
#12

yscarlet committed:
feat: post 기능 구현

yunuo46 commented on src/main/java/com/gaplog/server/domain/post/application/PostService.java:
```diff
@@ -0,0 +1,61 @@
+        return postRepository.save(post);
+    }
+
+    @Transactional
```
벌써 트랜잭션까지 적용하셨군요! 단순 조회 api같은 경우에는 readOnly 옵션을 고려해보셔도 좋을 것 같습니다

yunuo46 reviewed (commented):
수고하셨습니다! 질문은 마음껏 해주시고 api명세까지 완료되면 머지하겠습니다. 답변을 간단히 드리면,  dto의 필드가 동일하더라도 라이프사이클이 다르다면 각각 분류하는 편이 dto의 목적과 유지보수 측면에서 좋습니다. 형태가 아닌  라이프사이클에서의 유사성을 판단하여 동일한 dto를 사용하거나, 상속을 활용하시면 됩니다!
추가적으로 Entity의 경우에는 @MappedSuperclass 를 사용하기도 합니다~

yscarlet committed:
feat:api 및 service기능 추가

yscarlet committed:
fix: PostRepository 오류 수정

yscarlet committed:
add: category package 추가

yscarlet committed:
feat!: category type 변경

domain - Post의 type String -> Category 변경
dao - PostRepository의 findByCategoryId 주석처리
	=> 추후 수정 예정
application - PostService의createPost 수정
		=> 변경된 domain에 맞추어 변경
api - PostApi - putPost수정
		=> createPost 변경됨에 따라 맞추어 수정
dto - request - PostRequest수정
		=> category의 자료형을 기존의 string에서 Long(id)로 수전

yscarlet committed:
feat: PostService deletePost 기능로직 추가

Service에 기존에 없던 deletePost 기능 로직을 추가

yscarlet committed:
feat!: 카테고리별 조회 로직 변경

PostService/getFindByCategoryPostInfo 로직 변경
	Return type:  FindByCategoryPostResponse -> List<FindByCategoryPostResponse>
PostRrepository/findByCategoryId parameter 및 return type 변경
	Param: String -> Long
	Return type: Optional<Post> -> List<Post>
PostApi/getCategoryPost 로직 추가

yscarlet committed:
feat!: 키워드 조회 로직 변경

PostService/getFindByKeywordPostInfo 로직 변경
	Return type:  FindByKeywordPostResponse -> List<FindByKeywordPostResponse>
PostRrepository/findByKeyword 삭제 및 List<Post> findByTitleContainingOrContentContaining(String titleKeyword, String contentKeyword) 추가
PostApi/getKeywordPost 로직 추가

yunuo46 committed:
Merge branch 'main' into main
